src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx

```diff
@@ -124,17 +124,87 @@ class CampaignCannedResponsesForm extends React.Component<InnerProps, State> {
     }
   };
 
+  cannedResponseTitleSort = (
+    { title: titleFirst }: CannedResponse,
+    { title: titleSecond }: CannedResponse
+  ) => {
+    if (titleFirst > titleSecond) {
+      return 1;
+    }
+    if (titleFirst < titleSecond) {
+      return -1;
+    }
+    return 0;
+  };
+
+  cannedResponseArraysEqual = (
+    cannedResponses1: CannedResponse[],
+    cannedResponses2: CannedResponse[]
+  ) =>
+    cannedResponses1.length === cannedResponses2.length &&
+    cannedResponses1.every(
+      (cr, idx) => cr.title === cannedResponses2[idx].title
+    );
+
+  mapToTitleDisplayOrder = (cannedResponses: CannedResponse[]) => {
+    return cannedResponses.map((cannedResponse, idx) => {
+      return { ...cannedResponse, displayOrder: idx };
+    });
+  };
+
   handleOnSaveResponse = (response: CannedResponse) => {
     const newId = Math.random()
       .toString(36)
       .replace(/[^a-zA-Z1-9]+/g, "");
 
     const { cannedResponses } = this.pendingCannedResponses();
-    const cannedResponsesToAdd = this.state.cannedResponsesToAdd.concat({
+    const cannedResponsesAlphabetical = cannedResponses.sort(
+      this.cannedResponseTitleSort
+    );
+
+    let newCannedResponse = {
       ...response,
       id: newId,
       displayOrder: cannedResponses.length + 1
-    });
+    };
+
+    if (
```

Comment:
Because I wanted this to be more like a generic function for any `cannedResponse` arrays (ex. comparing the pending canned responses to the edited ones) I was thinking it was a good thing to include to check for weirdness. For the specific way it's currently being used, they should always have the same length